src/modules/notification/infrastructure/store/notification.store.ts

```diff
@@ -1,22 +1,33 @@
 import EventNotification from '../../domain/entity/EventNotification';
-import valtioEventNotificationStore, { EventNotificationState, subscribe } from './valtio/notification.store.valtio';
+import { store } from './modules/notification';
+import { ActionTypes, MutationTypes } from './types';
+import { MutationPayload } from 'vuex';
 
-export type StoreListener = (state: EventNotificationState) => void;
+export type StoreListener = (mutation: MutationPayload) => void;
 
 class EventNotificationStore {
   subscribeToEventNotifications(listener: StoreListener) {
-    subscribe(valtioEventNotificationStore, () => {
-      console.log(valtioEventNotificationStore);
-      listener(valtioEventNotificationStore);
+    store.subscribe((mutation: MutationPayload) => {
+      if (mutation.type === MutationTypes.SET_NOTIFICATIONS) {
+        listener(mutation.payload);
+      }
+    });
+  }
+
+  subscribeToLoadingState(listener: StoreListener) {
+    store.subscribe((mutation: MutationPayload) => {
+      if (mutation.type === MutationTypes.SET_LOADING) {
+        listener(mutation.payload);
+      }
```

Comment:
```suggestion
  subscribeToEventNotifications(listener: StoreListener) {
    store.subscribe((mutation: MutationPayload, state // find the correct type here) => {
      if (mutation.type === MutationTypes.SET_NOTIFICATIONS || mutation.type === MutationTypes.SET_LOADING) {
        listener(state. // find the correct slice of state here);
      }
    });
  }
```

I think you should do it this way to avoid having too many subscribe functions.